src/lib/repositories/impl_oauth2/oauth2_repository_impl.py

```diff
@@ -0,0 +1,195 @@
+from injector import inject
+from sqlalchemy import text
+from sqlalchemy.engine.base import Engine
+import jwt
+from datetime import datetime, timedelta
+
+from src.constants.http_status_code import HttpStatus
+from src.env_config import get_env_config_instance
+from src.utils.sql_oath_queries import (
+    SQL_QUERY_TO_GET_USER_BY_USER_NAME,
+    SQL_QUERY_TO_GET_CLIENT_BY_CLIENT_ID,
+    SQL_QUERY_TO_GET_CLIENT_BY_ID,
+    SQL_QUERY_TO_ADD_ACCESS_TOKEN,
+    SQL_QUERY_TO_ADD_REFRESH_TOKEN,
+    SQL_QUERY_TO_GET_REFRESH_TOKEN_BY_USER_NAME,
+    SQL_QUERY_TO_GET_ACCESS_TOKEN_BY_TOKEN,
+    SQL_QUERY_TO_GET_REFRESH_TOKEN_BY_ID,
+    SQL_QUERY_TO_DELETE_ACCESS_TOKEN,
+    SQL_QUERY_TO_DELETE_REFRESH_TOKEN,
+)
+
+
+class Oauth2Services:
+    @inject
+    def __init__(self, engine: Engine):
+        self.env_config = get_env_config_instance()
+        self.engine = engine
+
+    def login(self, user_name, password, client_id):
+        user = self._get_user_by_user_name(user_name)
+
+        if user:
+            if user.user_name == user_name and user.password == password:
```

Comment:
Use a mechanism to encrypt the password

Reply:
Done. In the services a use the `bcrypt.checkpw` method to compare password with the hash. The method to encrypt
passwords was create in the util but it is not used it yet. this method will be use when user services be created.